docs/docs/helm.mdx

```diff
@@ -0,0 +1,186 @@
+library.  
+The following example shows how to extract the individual resources of the
+[`grafana`](https://hub.helm.sh/charts/grafana/grafana) Helm Chart:
+
+```jsonnet
+local helm = (import "github.com/grafana/jsonnet-libs/helm-util/helm.libsonnet").new(std.thisFile);
+
+{
+  grafana: helm.template("grafana", "./charts/grafana", {
+    namespace: "monitoring",
+    values: {
+      persistence: { enabled: true }
+    }
+  })
+}
+```
+
+The Chart itself is required to be [vendored at a relative
+path](#vendoring-helm-charts), in this case `./charts/grafana`.
+
+> **Important:** You MUST include the `.new(std.thisFile)` part in the import.
+> This is what tells Tanka where you actually call `helm.template()` from, so it
+> can find your vendored Charts.
+
+<br />
+
+Once called, the `$.grafana` key holds the individual resources of Helm Chart as
+a regular Jsonnet object that looks roughly like so:
+
+```jsonnet
+{
+  cluster_role_binding_grafana_clusterrolebinding: {/* ... */},
+  cluster_role_grafana_clusterrole: {/* ... */},
+  config_map_grafana: {/* ... */},
+  config_map_grafana_test: {/* ... */},
+  deployment_grafana: {/* ... */},
+  // ...
+}
+```
+
+## Vendoring Helm Charts
+
+A core concept of Grafana Tanka is purity, which it **always yields the same
+resources**, regardless of surroundings. This is the case, when the project is
+strictly self-contained.
+
+Helm however keeps Charts and repository configuration somewhere around
+`~/.config/helm`, which violates above requirement.
+
+As such, we decided to only allow Helm Charts that can be found **inside the
+bounds of a project**. This means, you MUST put your Charts **somewhere next to
+the file that calls `helm.template()`**, so that it can be referred to using a
+relative path.
+
+### Vendor Location
+
+Where to actually put them inside the project is up to you, but keep in mind you
+need to refer to them using relative paths.
+
+We recommend always writing libraries that wrap the actual Helm Chart, so the
+consumer does not need to be aware of it. Whether you put these into `lib/` or
+publish them on GitHub (`vendor/`) is up to you.
+
+A library usually looks like this:
+
+```
+  /jsonnetfile.json
+  /main.libsonnet
+```
+
+When adopting Helm inside it, we recommend vendoring at the top level, as such:
+
+```diff
+  /jsonnetfile.json
+  /main.libsonnet
++ /charts
++ /charts/<someChart>
+```
+
+This way, you can refer to the charts as `./charts/<someChart>` from inside
+`main.libsonnet`. By keeping the Chart as close to the consumer as possible, the
+library is kept portable.
+
+### Charttool
+
+Helm does not make vendoring incredibly easy. `helm pull` provides the required
+plumbing, but it does not record its actions in a reproducible manner.
+
+Tanka ships a special utility at `tk tool charts`, which automates `helm pull`:
+
+```bash
+# Create a chartfile.yaml in the current directory, e.g. in lib/myLibrary
```

Comment:
nit: If we use `console` for highlighting, `#` denotes a root shell rather than a comment, technically `$ # comment` would be a comment.
```suggestion
$ # Create a chartfile.yaml in the current directory, e.g. in lib/myLibrary
```

Reply:
Not sure how I feel about this one. We use `bash` throughout our docs, and imo it renders just fine on tanka.dev, which is what matters.

Reply:
Yeah that's fair. I'm not sure that console provides anything valuable unless you need a difference in highlighting for command output which this doesn't have. It may also GitHub Markdown specific.